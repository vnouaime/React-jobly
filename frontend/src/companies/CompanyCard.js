import React from "react"
import { Link } from "react-router-dom";
import "./CompanyCard.css";

/** Shows some information about company.
 *
 * Rendered by company list to display card for each company. 
 *
 * CompanyList -> CompanyCard
 */

const CompanyCard = ({name, description, logoUrl, handle}) => {
    return (
        <Link className="CompanyCard card" to={`/companies/${handle}`}>
          <div className="card-body">
            <h6 className="card-title">
              {name}
              {logoUrl && <img src={logoUrl}
                               alt={name}
                               className="float-right ml-5" />}
            </h6>
            <p><small>{description}</small></p>
          </div>
        </Link>
    );
}

export default CompanyCard