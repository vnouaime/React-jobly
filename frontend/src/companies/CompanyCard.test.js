import React from "react";
import { render } from "@testing-library/react";
import CompanyCard from "./CompanyCard";
import { MemoryRouter } from "react-router";

it("matches snapshot with logo", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <CompanyCard
            handle="rithm"
            name="Rithm School"
            description="Become an exceptional developer in 16 weeks."
            logo_url="https://pbs.twimg.com/profile_images/770491761412173826/ZUeIa4tw_400x400.jpg"
        />
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot without logo", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <CompanyCard
            handle="test"
            name="test"
            description="testing"
        />
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});