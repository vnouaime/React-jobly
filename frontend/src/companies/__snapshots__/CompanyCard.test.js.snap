// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with logo 1`] = `
<DocumentFragment>
  <a
    class="CompanyCard card"
    href="/companies/rithm"
  >
    <div
      class="card-body"
    >
      <h6
        class="card-title"
      >
        Rithm School
      </h6>
      <p>
        <small>
          Become an exceptional developer in 16 weeks.
        </small>
      </p>
    </div>
  </a>
</DocumentFragment>
`;

exports[`matches snapshot without logo 1`] = `
<DocumentFragment>
  <a
    class="CompanyCard card"
    href="/companies/test"
  >
    <div
      class="card-body"
    >
      <h6
        class="card-title"
      >
        test
      </h6>
      <p>
        <small>
          testing
        </small>
      </p>
    </div>
  </a>
</DocumentFragment>
`;
